examples/api-routes-rate-limit/utils/rate-limit.js

```diff
@@ -0,0 +1,31 @@
+const LRU = require('lru-cache')
+
+const rateLimit = (options) => {
+  const tokenCache = new LRU({
+    max: parseInt(options.uniqueTokenPerInterval || 500, 10),
+    maxAge: parseInt(options.interval || 60000, 10),
+  })
+
+  return {
+    check: (res, limit, token) =>
+      new Promise((resolve, reject) => {
+        const tokenCount = tokenCache.get(token) || [0]
+        if (tokenCount[0] === 0) {
+          tokenCache.set(token, tokenCount)
+        }
+        tokenCount[0] += 1
```

Comment:
An observation for other passers-by looking to implement something similar. 😅 

This bit looks a little hacky to me. _(To the point where I don't understand how using `get` rather than `peek` doesn't actually seem to break this, while it kind of should?)_ If we're avoiding updating the "recently used"-ness of the values, wouldn't a plain `Map` work, given that the elements are in insertion order? Looks like we're asking an LRU cache to be a FIFO cache (with some timestamp-related behavior on `get` and `set`). Something like the following might be enough for this use-case?

```typescript
type WithTimestamp<Value> = { value: Value, timestamp: number }
type Options = { max: number, maxAge: number }

const Cache = <Key, Value>(options: Options) => {
  const map = new Map<Key, WithTimestamp<Value>>()

  return {
    get: (key: Key): Value | undefined => {
      const element = map.get(key)
      if (!element) return undefined
      if (Date.now() - element.timestamp > options.maxAge) {
        map.delete(key)
        return undefined
      }
      return element.value
    },
    set: (key: Key, value: Value): void => {
      if (map.size === options.max) map.delete(map.keys().next().value)
      const timestamp = map.get(key)?.timestamp ?? Date.now()
      map.set(key, { value, timestamp })
    },
    clear: (): void => map.clear(),
    size: (): number => map.size
  }
}
```